staging/src/kubevirt.io/api/core/v1/schema.go

```diff
@@ -729,8 +729,9 @@ type Disk struct {
 
 // CustomBlockSize represents the desired logical and physical block size for a VM disk.
 type CustomBlockSize struct {
-	Logical  uint `json:"logical"`
-	Physical uint `json:"physical"`
+	Logical            uint  `json:"logical,omitempty"`
+	Physical           uint  `json:"physical,omitempty"`
+	DiscardGranularity *uint `json:"discardGranularity,omitempty"`
```

Comment:
Is there any reason why we can't always detect this and "do the right thing?"

It is my understanding that autodetect does not always work with logical/physical block sizes and should be used cautiously because it is possible (likely in fact) that a bootable disk image only supports 512 lbs. But I don't think there should be an issue with discard

Reply:
Live migration compatibility requires a DiscardGranularity value that works on both the source and the destination. The risk of problems is greatest when migrating without shared storage because the source and destination disks could be from different vendors/models with different discard granularities.

This is why QEMU does not automatically expose the value to guest and why KubeVirt should also be careful about automatically exposing it. Leaving the decision to the user is inconvenient but ensures they have thougth about live migration compatibility.

Reply:
@stefanha I can understand why qemu cannot handle this but in kubevirt the target virt-launcher can detect the target discard size when initializing

Reply:
@kevmw okay thanks for clarifying, my assumption was that discard size was fixed from guest perspective when the VM starts (or disk is hotplugged) but qemu could translate between guest/host discard sizes when storage is migrated